solution

serentinos opened the pull request.

serentinos committed:
add solution for averageMenAge

serentinos committed:
add rounding to menAverageAge

serentinos committed:
add averagewoman solution

serentinos committed:
add working children finding and getting of age diff

serentinos committed:
add solution

serentinos committed:
refactor

33Orange commented on src/averageAges.js:
```diff
@@ -20,6 +20,21 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
```
Clean up the code from unnecessary comments.

33Orange commented on src/averageAges.js:
```diff
@@ -20,6 +20,21 @@ function calculateMenAverageAge(people, century) {
+    });
+
+  const averageMenAge = mens
+    .reduce((prev, { born, died }) => prev + (died - born), 0) / mens.length
```
You do the same calculations with reduce in every function. Maybe we should move this calculation to a separate helper?

33Orange commented on src/averageAges.js:
```diff
@@ -31,13 +46,25 @@ function calculateMenAverageAge(people, century) {
+    : people
+      .filter(person => person.sex === 'f');
+
+  const averageAge = filteredWoman
+    .reduce((sum, { born, died }) => sum + (died - born), 0)
+      / filteredWoman.length;
```
You do the same calculations with reduce in every function. Maybe we should move this calculation to a separate helper?

33Orange commented on src/averageAges.js:
```diff
@@ -49,13 +76,33 @@ function calculateWomenAverageAge(people, withChildren) {
+      .push(...ageDifference);
+  });
+
+  const avgDifference = differences
+    .reduce((sum, current) => sum + current, 0)
+    / differences.length;
```
You do the same calculations with reduce in every function. Maybe we should move this calculation to a separate helper?

33Orange commented on src/averageAges.js:
```diff
@@ -31,13 +46,25 @@ function calculateMenAverageAge(people, century) {
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWoman = withChildren
+    ? people
+      .filter(person => (
+        people.find(child => child.mother === person.name && person.sex === 'f')
+      ))
+    : people
+      .filter(person => person.sex === 'f');
```
You use a filter in both cases. Maybe it's better to do a check `withChildren` condition inside the filter?

33Orange commented on src/averageAges.js:
```diff
@@ -49,13 +76,33 @@ function calculateWomenAverageAge(people, withChildren) {
+  const differences = [];
+
+  people.forEach(person => {
+    const personChildren = onlyWithSon
+      ? people
+        .filter(child => person.name === child.mother && child.sex === 'm')
+      : people
+        .filter(child => person.name === child.mother);
```
You use a filter in both cases. Maybe it's better to do a check `onlyWithSon` condition inside the filter?

33Orange reviewed (changes requested):
Well done!

A few moment with readability

serentinos committed:
refactor code, remove redutant methods
- reduce now accepts only arrays with numbers, not objects
- remove redutant filter methods, now ternary are inside of filter method

serentinos committed:
average age calculation moved to a separate function

serentinos commented:
@33Orange Thank you for all your suggestions! This advice is really helpful!

33Orange reviewed (approved):
Well done ! 🔥
